### ISSUE TITLE
Support redirection rule with yaml source file

### ISSUE THREAD

**Issue by fenxu** [dotnet/docfx#3938]
**DocFX Version Used**: 
v3

Feature requirement:
Redirection rule should also support source file with yml extension.

**Comment by fenxu** [comment]
cc @herohua

**Comment by OsmondJiang** [comment]
does that mean we should support `a.yml`: 'xxx/xxx' in `redirection.json` files in v3? if yes, we already supported it.

**Comment by fenxu** [comment]
@OsmondJiang Yes, that's it. If it has been done, then please also make sure that the document id is also calculated correctly.
